scripts/aws_caper_server/create_instance.sh

```diff
@@ -0,0 +1,281 @@
+sudo mkdir -p $CAPER_CONF_DIR/local_loc_dir $CAPER_CONF_DIR/local_out_dir
+
+### make caper conf file
+cat <<EOF > $GLOBAL_CAPER_CONF_FILE
+# caper
+backend=aws
+no-server-heartbeat=True
+# cromwell
+max-concurrent-workflows=300
+max-concurrent-tasks=1000
+# local backend
+local-out-dir=$CAPER_CONF_DIR/local_out_dir
+local-loc-dir=$CAPER_CONF_DIR/local_loc_dir
+# aws backend
+aws-batch-arn=$AWS_BATCH_ARN
+aws-region=$AWS_REGION
+aws-out-dir=$AWS_OUT_DIR
+aws-loc-dir=$AWS_LOC_DIR
+# metadata DB
+db=postgresql
+postgresql-db-ip=$POSTGRESQL_DB_IP
+postgresql-db-port=$POSTGRESQL_DB_PORT
+postgresql-db-user=$POSTGRESQL_DB_USER
+postgresql-db-password=$POSTGRESQL_DB_PASSWORD
+postgresql-db-name=$POSTGRESQL_DB_NAME
+EOF
+sudo chmod +r $GLOBAL_CAPER_CONF_FILE
+
+### soft-link conf file for root
+sudo mkdir -p $ROOT_CAPER_CONF_DIR
+sudo ln -s $GLOBAL_CAPER_CONF_FILE $ROOT_CAPER_CONF_DIR
+
+### caper conf shared with all users
+sudo touch $AWS_AUTH_SH
+echo \"mkdir -p ~/.caper\" >> $AWS_AUTH_SH
+echo \"ln -s /opt/caper/default.conf ~/.caper/ 2> /dev/null | true\" >> $AWS_AUTH_SH
+"""
+
+# append more init commands to the startup-script
+STARTUP_SCRIPT="""$STARTUP_SCRIPT
+### init PostgreSQL for Cromwell
+sudo -u postgres createuser root -s
+sudo createdb $POSTGRESQL_DB_NAME
+sudo psql -d $POSTGRESQL_DB_NAME -c \"create extension lo;\"
+sudo psql -d $POSTGRESQL_DB_NAME -c \"create role $POSTGRESQL_DB_USER with superuser login password '$POSTGRESQL_DB_PASSWORD'\"
+
+### upgrade pip and install caper croo
+sudo python3 -m pip install --upgrade pip
+sudo -H pip3 install --ignore-installed PyYAML
+sudo pip install caper croo
+"""
+
+echo "$(date): Making a temporary startup script..."
+echo "$STARTUP_SCRIPT" > tmp_startup_script.sh
+
+# find the most recent AMI matching the name search query
+# https://gist.github.com/vancluever/7676b4dafa97826ef0e9
+echo "$(date): Searching for AMI with name matching \"${AMI_NAME_SEARCH_QUERY}\" in region ${AWS_REGION}..."
+AMI=$(aws --region "${AWS_REGION}" ec2 describe-images --filters "Name=name,Values=${AMI_NAME_SEARCH_QUERY}" --query 'sort_by(Images,&CreationDate)[-1].ImageId')
+AMI="${AMI%\"}"
+AMI="${AMI#\"}"
+echo "$(date): Found AMI: ${AMI}"
+
+echo "$(date): Creating an instance..."
+aws ec2 --region "${AWS_REGION}" run-instances \
+  --tag-specifications "ResourceType=instance,Tags=[{Key=Name,Value=${INSTANCE_NAME}}]" \
+  --image-id "${AMI}" \
+  --subnet-id "${PUBLIC_SUBNET_ID}" \
+  --key-name "${KEY_PAIR_NAME}" \
+  --block-device-mappings "DeviceName=${BOOT_DISK_DEVICE_NAME},Ebs={VolumeSize=${BOOT_DISK_SIZE}}" \
+  --instance-type "${INSTANCE_TYPE}" \
+  --user-data "file://tmp_startup_script.sh"
+echo "$(date): Created an instance successfully."
+
+echo "$(date): Deleting the temporary startup script..."
+rm -f tmp_startup_script.sh
+
+echo "$(date): Go to the AWS console and change instance's secrutiy group settings to allow SSH connection (port 22) from your IP (or IP range)."
```

Comment:
Thanks for the suggestion. I thought about it too and could do this by using `aws ec2 authorize-security-group-ingress` but decided to let users change it manually so that they are more careful about their security settings. I just deleted this line not to confuse users. It is already described in `README.md`.